presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -188,4 +189,25 @@ public void testCastFromVarchar()
         assertFunction("cast('abc' as JSON)", JSON, "\"abc\"");
         assertFunction("cast('\"a\":2' as JSON)", JSON, "\"\\\"a\\\":2\"");
     }
+
+    @Test
+    public void testIndeterminate()
+    {
+        assertOperator(INDETERMINATE, "cast(JSON '128' as BIGINT)", BOOLEAN, false);
```

Comment:
These tests don't check the operator for the JSON type, but for the specific type the values are being cast to.